src/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -66,4 +66,17 @@
   <ItemGroup>
     <ReferenceFromRuntime Include="System.Private.CoreLib" />
   </ItemGroup>
+
+  <Target Name="EnsureResourceDirectoryExists" BeforeTargets="_BatchGenerateResxSource">
+    <MakeDir Directories="$(IntermediateOutputPath)" Condition="!Exists($(IntermediateOutputPath))" />
+  </Target>
+
+  <!-- The Source Package will call into this target in order to retrieve the compiled resources file as
+       it will need to be part of the source package contents. -->
+  <Target Name="GetCompiledResourceFile" Returns="$(CompiledResourceFileForProject)" DependsOnTargets="_GenerateResxSource">
```

Comment:
Did you consider making this project & target responsible for flowing the sources, rather than making the PkgProj do that?  That way you could simply invoke this target and collect its output in the pkgproj.

Reply:
Don't depend on _GenerateResxSource, either depend on CoreCompile or expose a public target to depend on.

Reply:
> That way you could simply invoke this target and collect its output in the pkgproj.

That's exactly what I do. In the pkgproj all I do is to call into this target in order to flow the sources over.

https://github.com/dotnet/corefx/blob/edeaa0c00ff2adc9a063f05e01a35dbb537dab53/src/System.Text.Json/pkg/Microsoft.Bcl.Json.Sources.pkgproj#L25-L29